### PR TITLE
Paging and sorting

### PR DESCRIPTION
**Pull request by carolinaWolox**
## Summary
- Paging and sorting
## Trello Card
https://trello.com/c/Gxm1a6va/24-paging-and-sorting-repository

### CODE REVIEW

**Review by julianvasquez-wolox — APPROVED** [review]
Good job! I'll approve this but remember to check the comments on the previous two PR

File: src/main/java/wolox/training/controllers/BookController.java [hunk 1]
```diff
@@ -70,10 +75,55 @@ public String greeting(@RequestParam(name = "name", required = false, defaultVal
         return bookRepository.findByTitle(bookTitle);
     }
 
-    @GetMapping("/publisherGenreYear/{publisher}/{genre}/{year}")
-    public List<Book> findByGenrePublisherYear(@PathVariable String publisher, @PathVariable String genre,
-            @PathVariable String year) {
-        return bookRepository.findByPublisherAndGenreAndYear(publisher, genre, year);
+    @GetMapping("/publisherGenreYear")
```
Comment (juanwongramirez) [inline src/main/java/wolox/training/controllers/BookController.java:79]: you can delete this endpoint and use the @GetMapping("/allBooks") that endpoint uses the same params of this

File: src/main/java/wolox/training/controllers/BookController.java [hunk 2]
```diff
@@ -70,10 +75,55 @@ public String greeting(@RequestParam(name = "name", required = false, defaultVal
...
+    public Page<Book> findByGenrePublisherYear(@RequestParam(required = false) String publisher,
+            @RequestParam(required = false) String genre,
+            @RequestParam(required = false) String year,
+            @PageableDefault(sort = {Constants.SORT_BOOK}, value = 5) Pageable pageable) {
+
+        return bookRepository.findByPublisherAndGenreAndYear(publisher, genre, year, pageable);
+    }
+
+    @GetMapping("/allBooks")
```
Comment (juanwongramirez) [inline src/main/java/wolox/training/controllers/BookController.java]: You can replace @GetMapping("/allBooks") to @GetMapping("/"), finally this endpoint returns all values if all params are null

**Review by juanwongramirez — CHANGES_REQUESTED** [review]

File: src/main/java/wolox/training/controllers/BookController.java (hunk 2 above)
Comment (carolinaWolox) [reply src/main/java/wolox/training/controllers/BookController.java]: Fixed at last commit!

File: src/main/java/wolox/training/controllers/BookController.java (hunk 1 above)
Comment (carolinaWolox) [reply src/main/java/wolox/training/controllers/BookController.java:79]: Fixed at last commit!

**Review by juanwongramirez — APPROVED** [review]

**Pull request merged**